operators/src/pro/cache/cache_operator.rs

```diff
@@ -108,43 +157,51 @@ where
 }
 
 #[async_trait]
-impl<Q, T> QueryProcessor for CacheQueryProcessor<Q, T>
+impl<P, S, E, Q> QueryProcessor for CacheQueryProcessor<P, E, Q>
 where
-    Q: QueryProcessor<Output = RasterTile2D<T>, SpatialBounds = SpatialPartition2D> + Sized,
-    T: Pixel + Cachable,
+    P: QueryProcessor<Output = E, SpatialBounds = Q> + Sized,
+    E: CacheElement<CacheElementSubType = S, Query = QueryRectangle<Q>>
+        + ResultStreamWrapper
+        + Send
+        + Sync
+        + Clone
+        + 'static,
+    S: CacheElementSubType<CacheElementType = E> + Send + Sync + 'static,
+    Q: AxisAlignedRectangle + Send + Sync + 'static,
+    E::ResultStream: Stream<Item = Result<E>> + Send + Sync + 'static,
+    SharedCache: AsyncCache<E>,
 {
-    type Output = RasterTile2D<T>;
-    type SpatialBounds = SpatialPartition2D;
+    type Output = E;
+    type SpatialBounds = Q;
 
     async fn _query<'a>(
         &'a self,
         query: QueryRectangle<Self::SpatialBounds>,
         ctx: &'a dyn QueryContext,
     ) -> Result<BoxStream<'a, Result<Self::Output>>> {
-        let tile_cache = ctx
+        let shared_cache = ctx
             .extensions()
-            .get::<Arc<TileCache>>()
+            .get::<Arc<SharedCache>>()
```

Comment:
can you adjust the expect message below? It still says TileCache

Reply:
[865f648](https://github.com/geo-engine/geoengine/pull/833/commits/865f6487735022b59e5ca967c325e45d832161d1)